project-icp/bee-pollinator-app
# Add Yield Results Chart

# Overview
Up until now we've been `console.debug()`ing the yield model results and letting the old MMW tr55 runoff charts error out in the UI. At last, this PR removes the tr55 charts and adds in a yield chart!

The yield chart displays the yield per acre for each crop type.
- On scenario views,  we show the scenario's yield and the current conditions' yield for comparison. 
- On current conditions we just show the current conditions
- We only show yield for crops with either current conditions or scenario yields not equal to zero.

#### Notes
Typically grouped bar charts distinguish each group member/series by color, and then label what each is in a legend. Our bar chart, however, has each crop have its own color, and only distinguishes the series (scenario/current conditions) by shade (really fill opacity). This makes it difficult to label in the legend because there's no unifying color for all the current conditions bars, or all the scenario bars. For now, the tooltip gives the series name, but which bar is which isn't otherwise readily apparent. Should we add a pattern to one of the series so that we can add a legend, or maybe add additional labels to the bars or something?

# Demo

#### Current conditions shows single bar per crop
![screen shot 2016-12-29 at 12 02 52 pm](https://cloud.githubusercontent.com/assets/7633670/21549351/a0a83f48-cdbf-11e6-8e25-9cfb38a7ace0.png)

#### Scenario shows current conditions and scenario results
![screen shot 2016-12-29 at 12 03 20 pm](https://cloud.githubusercontent.com/assets/7633670/21549349/a0a3355c-cdbf-11e6-878a-8753bc730754.png)

#### Tooltip gives value and whether bar is current Conditions/scenario
<img width="627" alt="screen shot 2016-12-29 at 12 03 31 pm" src="https://cloud.githubusercontent.com/assets/7633670/21549350/a0a4e83e-cdbf-11e6-9ae0-98a67570c747.png">

# Testing Instructions
Pull this branch and `./scripts/bundle.sh --debug`
- Open dev tools - as you go through confirm there are no debug statements or errors
- Draw an AOI and get to the model view
   - most places will show "No Data"
   - `Drummond Dr, Yuba City` will have orchard/almond values
- Add land cover in current conditions and in a scenario
  - confirm the chart updates and shows different colors + appropriate labels for each crop type

Connects #89
